include/acl_selectors.php

```diff
@@ -625,31 +631,29 @@ function _contact_link($i){ return dbesc($i['link']); }
 				intval($conv_id),
 				dbesc($search),
 				dbesc($search),
-				implode("','", $known_contacts)
+				implode("', '", $known_contacts)
 		);
-		if (dbm::is_result($r)){
+		if (dbm::is_result($r)) {
 			foreach ($r as $row) {
-				// nickname..
-				$up = parse_url($row['author-link']);
-				$nick = explode("/",$up['path']);
-				$nick = $nick[count($nick)-1];
-				$nick .= "@".$up['host'];
-				// /nickname
-				$unknow_contacts[] = array(
-					'type'    => 'c',
-					'photo'   => proxy_url($row['author-avatar'], false, PROXY_SIZE_MICRO),
-					'name'    => htmlentities($row['author-name']),
-					'id'      => '',
-					'network' => 'unknown',
-					'link'    => $row['author-link'],
-					'nick'    => htmlentities($nick),
-					'forum'   => false
-				);
+				$contact = get_contact_details_by_url($row['author-link'], 0);
+
+				if (count($contact) > 0) {
+					$unknown_contacts[] = array(
+						'type'    => 'cu',
+						'photo'   => proxy_url($contact['micro'], false, PROXY_SIZE_MICRO),
+						'name'    => htmlentities($contact['name']),
+						'id'      => intval($contact['id']),
+						'network' => $contact['network'],
+						'link'    => $contact['url'],
+						'nick'    => $contact['nick'],
```

Comment:
I just saw that the old ```$nick``` includes the host part. So ```address``` would be the correct array element. Additionally you may check if this array element has an value.